ceno_zkvm/src/instructions/riscv/mul.rs

```diff
@@ -319,6 +357,20 @@ impl<E: ExtensionField, I: RIVInstruction> Instruction<E> for MulhInstructionBas
                 let prod = rs1_val.as_u64() * rs2_val.as_u64();
                 (prod % (1u64 << BIT_WIDTH)) as u32
             }
+            MulhSignDependencies::LL { constrain_rd } => {
+                let prod = rs1_val.as_u64() * rs2_val.as_u64();
+                let prod_lo = (prod % (1u64 << BIT_WIDTH)) as u32;
```

Comment:
With `(prod % (1u64 << BIT_WIDTH)) as u32;` I need to understand two operations: the modulo and the cast.  That's strictly more complicated than just understanding a strict subset of these operations.

(This might only seem easier for you, because you understand that in our specific circumstances the cast kind-of doesn't do anything.  But to come to that conclusion that make understanding easier, you first have to understand the whole thing.  Which was the point in the first place..)

Reply:
fixed in [2114670](https://github.com/scroll-tech/ceno/pull/660/commits/2114670c5281259da340908eddecb07f29d4b608)